sdks/python/apache_beam/yaml/yaml_mapping.md

```diff
@@ -200,3 +200,40 @@ criteria. This can be accomplished with a `Filter` transform, e.g.
     language: sql
     keep: "col2 > 0"
 ```
+
+## Types
+
+Beam will try to infer the types involved in the mappings, but sometimes this
+is not possible. In these cases one can explicitly denote the expected output
+type, e.g.
+
+```
+- type: MapToFields
+  config:
+    language: python
+    fields:
+      new_col:
+        expression: "col1.upper()"
+        type: string
+```
+
+The expected type is given in json schema notation, with the addition that
+a top-level basic types may be given as a literal string rather than requiring
+a `{type: 'basic_type_name'}` nesting.
+
+```
+- type: MapToFields
+  config:
+    language: python
+    fields:
+      new_col:
+        expression: "col1.upper()"
+        type: string
+      another_col:
+        expression: "beam.Row(a=col1, b=[col2])"
+        type:
+          type: 'object'
+          properties:
+            a: {type: 'string'}
+            b: {type: 'array', items: {type: 'number'}}
```

Comment:
I like the idea of keeping the type properties as nested YAML rather than JSON
```suggestion
      another_col:
        expression: "beam.Row(a=col1, b=[col2])"
        type:
          type: 'object'
          properties:
            a: 
              type: 'string'
            b: 
              type: 'array'
              items: 
                type: 'number'
```
I also think naming the outer type tag as `output_type` or something similar could help make the start of the type config more explicit, but this might be too verbose. Also, why is the type `'object'`? Why not `'row'`?
```suggestion
      another_col:
        expression: "beam.Row(a=col1, b=[col2])"
        output_type:
          type: 'object'
          properties:
            ...
```

Reply:
It's "object" because I'm borrowing the JSON schema specification wholesale. 

Changing to `output_type`. It's a bit more verbose, but definitely clearer.